query/src/main/java/org/infinispan/query/clustered/DistributedIterator.java

```diff
@@ -69,6 +73,8 @@ class DistributedIterator<T> implements CloseableIterator<T> {
       } else {
          mergedResults = TopDocs.merge(firstResult, maxResults, partialTopDocs);
       }
+      batchSize = cache.getCacheConfiguration().clustering().stateTransfer().chunkSize();
```

Comment:
What happens if the chunkSize is > than the maxResults?
In this case I think that we can optimize the fetch to return the maxResults size.
What happens if the maxResults is > then the chankSize?
In this case do we fetch the other values expected by the current page?

Reply:
* chunkSize > maxResults

The method `hasMoreKeys()` is bounded by `maxResults` and it breaks the for loop before reaching the `chunkSize`.
The only downside is when chunkSize >> maxResults because I'm allocating an array way larger than it needs to be. I'll fix that as well.

* chunkSize < maxResults

when it finishes the current batch, it fetches the next `chunkSize` keys

```java
   public final boolean hasNext() {
      if (valueIndex == values.size()) {
         fetchBatch();
      }
      return valueIndex < values.size();
   }
```

Reply:
Great! I'm resolving it.